tapeagents/finetune/rl/__init__.py

```diff
@@ -210,6 +210,8 @@ def update_rewards_and_advantages(dataset: Dataset, config: RLConfig) -> Dataset
 
     """
     df = dataset.to_pandas()
+    group_ids = list(df['group_id'])
```

Comment:
Sorry, a bit of nitpicking, but it's better to use double quotes for all strings except interpolated inside other strings. You can run `make lint` and ruff will catch all stuff of that sort.